### ISSUE TITLE
Add mount 7z

### ISSUE THREAD

**Issue by AndyILC** [aldostools/webMAN-MOD#112]
Is it possible to add such functionality.
Mounting 7-zip files without compression.
Created for example by this command:
7z a -r -t7z -mx0
With this structure:
\BLES01976.ird
\BLES01976-[Minecraft PlayStation3 Edition]
\BLES01976-[Minecraft PlayStation3 Edition]\PS3_DISC.SFB
\BLES01976-[Minecraft PlayStation3 Edition]\PS3_GAME
.......
This will store everything, you needed in one archive, game files and IRD file.
And also be sure, that the data is not damaged.

**Comment by aldostools** [comment]
ISO files store all the game files in one archive as you suggest.

IRD support is not implemented in webMAN. However you can check the MD5 of the ISO to verify the integrity of the file.

**Comment by AndyILC** [comment]
Thanks for the answer!
You do not need to add any functionality to work with IRD.
I have 1200 such 7z files.
And I would like to mount them over the network.
Storage and periodic check of such quantity of ISO files is almost impossible.
In the form of 7z, you can check and additionally store for example IRD.
When stored in ISO, it is necessary to store separately and md5 and IRD and can store something else extra

**Comment by aldostools** [comment]
It's very simple to check the integrity of the ISO files:
1- Download any MD5 tool with directory support. e.g. I developed PS3 Game Integrity which is included in my PS3 Tools Collection (google it).
2- Scan the PS3ISO folder (or any other ISO folder) and store the MD5 hashes is a text file
3- Periodically repeat the scan and compare the resulting MD5 hashes. e.g. PS3 Game Integrity does the comparison automatically.

**Comment by AndyILC** [comment]
On points 1 2 3 I understand this perfectly.
And storage in ISO does not suit me at all.
Since there is no possibility to quickly replace eg 1 file if damaged.
It is not possible to add IRD, if it appeared.
etc.

Is it fundamentally impossible?
Or it can be implemented for example in the form of a plug-in?
When transmitting to anyone, the ISO does not have the ability to transmit all the additional information. And no one will check the quality of this ISO.

**Comment by aldostools** [comment]
It is "possible" to use zip or any format, if a developer with the skills takes the time and effort to do it...

However, it is not practical even if it is "possible". The file system in the PS3 read the games from a virtual bluray drive requesting data to be executed from sectors. This is the exact way how the data is stored in ISO. Using a different format would require to also create a map of sectors for each file, which is not efficient. Indeed ps3netsrv does it dynamically when a game folder is mounted. That's why it takes a while longer to mount a game folder.

In short: what you are requesting requires a lot of development and testing go do something that already works efficiently.

You can use an ISO editor if you need to modify the internal files. Or simply extract the files, modify and rebuild the ISO.

**Comment by AndyILC** [comment]
Can it be more convenient to add this functionality to ps3netsrv?
And do not make changes to the webman?
Is this possible?
Will this be enough?

**Comment by aldostools** [comment]
It makes more sense in ps3netsrv. Maybe someone else want to do it. I can't dedicate time to this.

**Comment by AndyILC** [comment]
I understood!
Thanks for your reply.
